vnext/Microsoft.ReactNative/Fabric/platform/react/renderer/textlayoutmanager/TextLayoutManager.cpp

```diff
@@ -5,75 +5,86 @@
-// actual view component. -- Any properties that rely on the context from the UI tree should be set directly on the
-// TextBlock here and in the view component.
 TextMeasurement TextLayoutManager::measure(
     AttributedStringBox attributedStringBox,
     ParagraphAttributes paragraphAttributes,
     LayoutConstraints layoutConstraints) const {
+  winrt::com_ptr<IDWriteFactory> spDWriteFactory;
+
+  if (!m_spDWriteFactory) {
+    DWriteCreateFactory(
+        DWRITE_FACTORY_TYPE_SHARED, __uuidof(IDWriteFactory), reinterpret_cast<IUnknown **>(m_spDWriteFactory.put()));
+  }
+
   for (auto &fragment : attributedStringBox.getValue().getFragments()) {
-    if (!m_textBlock) {
-      m_textBlock = xaml::Controls::TextBlock();
-    }
-
-    auto textblock = xaml::Controls::TextBlock();
-    m_textBlock.Text(winrt::to_hstring(fragment.string));
-    m_textBlock.FontSize(fragment.textAttributes.fontSize);
-    m_textBlock.FontWeight(
-        winrt::Windows::UI::Text::FontWeight{static_cast<uint16_t>(fragment.textAttributes.fontWeight.value_or(
-            static_cast<facebook::react::FontWeight>(DWRITE_FONT_WEIGHT_REGULAR)))});
-
-    switch (fragment.textAttributes.fontStyle.value_or(facebook::react::FontStyle::Normal)) {
-      case facebook::react::FontStyle::Italic:
-        m_textBlock.FontStyle(winrt::Windows::UI::Text::FontStyle::Italic);
-        break;
-      case facebook::react::FontStyle::Normal:
-        m_textBlock.FontStyle(winrt::Windows::UI::Text::FontStyle::Normal);
-        break;
-      case facebook::react::FontStyle::Oblique:
-        m_textBlock.FontStyle(winrt::Windows::UI::Text::FontStyle::Oblique);
-        break;
-      default:
-        assert(false);
-    }
-
-    if (fragment.textAttributes.fontFamily.empty())
-      m_textBlock.FontFamily(xaml::Media::FontFamily(L"Segoe UI"));
-    else
-      m_textBlock.FontFamily(
-          xaml::Media::FontFamily(Microsoft::Common::Unicode::Utf8ToUtf16(fragment.textAttributes.fontFamily)));
-
-    winrt::Windows::Foundation::Size availableSize(
-        layoutConstraints.maximumSize.width, layoutConstraints.maximumSize.height);
-    m_textBlock.Measure(availableSize);
-    TextMeasurement tm;
-    auto size = m_textBlock.DesiredSize();
+    DWRITE_FONT_STYLE style = DWRITE_FONT_STYLE_NORMAL;
+    if (fragment.textAttributes.fontStyle == facebook::react::FontStyle::Italic)
+      style = DWRITE_FONT_STYLE_ITALIC;
+    else if (fragment.textAttributes.fontStyle == facebook::react::FontStyle::Oblique)
+      style = DWRITE_FONT_STYLE_OBLIQUE;
+
+    winrt::com_ptr<IDWriteTextFormat> spTextFormat;
+    m_spDWriteFactory->CreateTextFormat(
+        fragment.textAttributes.fontFamily.empty()
+            ? L"Segoe UI"
+            : Microsoft::Common::Unicode::Utf8ToUtf16(fragment.textAttributes.fontFamily).c_str(),
+        NULL, // Font collection (NULL sets it to use the system font collection).
```

Comment:
>NULL [](http://example.com/codeflow?start=8&length=4)

I would consider NULL to be deprecated in most contexts. We better use nullptr or 0.